packages/shared/src/lib/shares.ts

```diff
@@ -163,7 +163,7 @@ export const processFileOnShareOrHttp = async <ProcessedDataResponse>(
 	ignoreCache: boolean,
   version: string,
   canProcessFileOnHttp?: boolean,
-	processFile: (mountedFilePath: string) => Promise<ProcessedDataResponse>
+	processFile: ({ file, cachePath }: { file: string, cachePath?: string }) => Promise<ProcessedDataResponse>
```

Comment:
Changed the type signature here to allow optional arguments to be passed to the function. Changing to named arguments also makes it easier to make modification to the function in the future.